### ISSUE TITLE
extendr-api cannot be compiled when ndarray is enabled

### ISSUE THREAD

**Issue by yutannihilation** [extendr/extendr#111]
As I commented on https://github.com/extendr/rextendr/issues/10#issuecomment-755387234, I see this error on my local, but this doesn't happen on CI runs. Does anyone see the same errors? (I'm using Rust 1.49 on Ubuntu 20.04).

``` console
❯ cargo test --manifest-path extendr-api/Cargo.toml --features ndarray
   Compiling libR-sys v0.2.0
   Compiling extendr-engine v0.1.11 (/home/yutani/repo/extendr/extendr-engine)
   Compiling extendr-api v0.1.11 (/home/yutani/repo/extendr/extendr-api)
error[E0106]: missing lifetime specifier
  --> extendr-api/src/robj_ndarray.rs:14:24
   |
14 |     Robj: AsTypedSlice<T>,
   |                        ^ expected named lifetime parameter
   |
help: consider using the `'a` lifetime
   |
14 |     Robj: AsTypedSlice<'a, T>,
   |                        ^^^

error[E0106]: missing lifetime specifier
  --> extendr-api/src/robj_ndarray.rs:14:24
   |
14 |     Robj: AsTypedSlice<T>,
   |                        ^ expected named lifetime parameter
   |
help: consider using the `'a` lifetime
   |
14 |     Robj: AsTypedSlice<'a, T>,
   |                        ^^^

error[E0310]: the parameter type `T` may not live long enough
  --> extendr-api/src/robj_ndarray.rs:17:31
   |
12 | impl<'a, T> FromRobj<'a> for ArrayView1<'a, T>
   |          - help: consider adding an explicit lifetime bound...: `T: 'static`
...
17 |         if let Some(v) = robj.as_typed_slice() {
   |                               ^^^^^^^^^^^^^^ ...so that the reference type `&[T]` does not outlive the data it points at

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0106, E0310.
For more information about an error, try `rustc --explain E0106`.
error: could not compile `extendr-api`
```

**Comment by clauswilke** [comment]
I get the same error on Rust 1.48.

**Comment by clauswilke** [comment]
And if I apply the fix suggested by the compiler (`Robj: AsTypedSlice<'a, T>`) the error goes away.

If this error doesn't arise on CI runs are we sure that this code actually gets compiled and tested during CI?

**Comment by yutannihilation** [comment]
Fixed by #118
